go/consensus/tendermint/abci/mux.go

```diff
@@ -574,6 +576,12 @@ func (mux *abciMux) EstimateGas(caller signature.PublicKey, tx *transaction.Tran
 		return 0, consensus.ErrInvalidArgument
 	}
 
+	// Certain modules, in particular the beacon require InitChain or BeginBlock
+	// to have completed before initialization is complete.
+	if mux.lastBeginBlock == blockHeightInvalid {
```

Comment:
Does this require a mutex/atomic since `EstimateGas` can be called in parallel to `BeginBlock`?

Reply:
think that variable is updated at the top of begin block too, kind of annoying